components/ai_chat/resources/page/chat_ui.tsx

```diff
@@ -57,6 +55,72 @@ function Content() {
   return <FullScreen />
 }
 
+function ConversationEntries(props: ConversationEntriesProps) {
+  const conversationContext = useConversation()
+  const iframeRef = React.useRef<HTMLIFrameElement|null>(null)
+  const hasNotifiedContentReady = React.useRef(false)
+  const [hasLoaded, setHasLoaded] = React.useState(false)
+
+  // Notify onIsContentReady when
+  // - iframe increases in height after a conversation change OR
+  // - iframe is loaded AND iframe conversation length is 0 after a conversation change
+
+  // Reset when conversation changes
+  React.useEffect(() => {
+    setHasLoaded(false)
+    props.onIsContentReady(false)
+    hasNotifiedContentReady.current = false
+    if (iframeRef.current) {
+      iframeRef.current.style.height = '0px'
+    }
+  }, [conversationContext.conversationUuid, props.onIsContentReady])
+
+  // The iframe has loaded if there're no conversation entries,
+  // it will never grow until a user action happens.
+  React.useEffect(() => {
+    // conversationUuid populated is a sign that data has been fetched
+    if (!hasNotifiedContentReady.current && conversationContext.conversationUuid &&
+        !conversationContext.conversationHistory.length && hasLoaded) {
+      hasNotifiedContentReady.current = true
+      props.onIsContentReady(true)
+    }
+  }, [
+    conversationContext.conversationUuid,
+    conversationContext.conversationHistory.length,
+    hasLoaded
+  ])
+
+  React.useEffect(() => {
+    const listener = (height: number) => {
+      // Use the first height change to notify that the iframe has rendered,
+      // in lieu of an actual "has rendered the conversation entries" event
+      // which, if we get any bugs with this and need to add complexity, might
+      // be simpler to implement excplicitly, from child -> parent.
+      if (!hasNotifiedContentReady.current && height > 0) {
+        hasNotifiedContentReady.current = true
+        props.onIsContentReady(true)
+      }
+      if (iframeRef.current) {
+        iframeRef.current.style.height = height + 'px'
+        props.onHeightChanged()
+      }
+    }
+    const id = api.conversationEntriesFrameObserver.childHeightChanged.addListener(listener)
+
+    return () => {
+      api.conversationEntriesFrameObserver.removeListener(id)
+    }
+  }, [props.onHeightChanged, props.onIsContentReady])
+
+  return (
+    <iframe
```

Comment:
let's put the `sandbox` attribute here to prevent it from doing weird stuff it doesn't need to do. it will need `allow-scripts` and probably `allow-same-origin`.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#sandbox

Reply:
Looks like it does need `allow-same-origin`